ui/components/component-library/checkbox/checkbox.scss

```diff
@@ -42,6 +42,7 @@
 
   &--disabled {
     opacity: var(--opacity-disabled);
+    cursor: not-allowed;
```

Comment:
Makes it so labels are not pointers

### Before

https://github.com/MetaMask/metamask-extension/assets/8112138/af878e83-82a3-4245-b91a-328a804b00e8

### After

https://github.com/MetaMask/metamask-extension/assets/8112138/13bf384a-c816-48e5-a8fe-481e3720ee45